Containers/talk/start.sh

```diff
@@ -16,43 +16,46 @@ elif [ -z "$INTERNAL_SECRET" ]; then
 fi
 
 set -x
-IPv4_ADDRESS_TALK="$(dig nextcloud-aio-talk A +short)"
+if [ -n "$(dig nextcloud-aio-talk A +short | grep -P "^[0-9.]+$" | head -n1)" ]; then
+    IPv4_ADDRESS_TALK="$(dig nextcloud-aio-talk A +short | grep -P "^[0-9.]+$" | head -n1)"
+fi
+if [ -n "$(dig nextcloud-aio-talk AAAA +short | grep -P "^[0-9a-fA-F:]+$" | head -n1)" ]; then
+    IPv6_ADDRESS_TALK="$(dig nextcloud-aio-talk AAAA +short | grep -P "^[0-9a-fA-F:]+$" | head -n1)"
+fi
+
+if [ -n "$(dig "$NC_DOMAIN" A +short | grep -P "^[0-9.]+$" | head -n1)" ]; then
+    IPv4_ADDRESS_NC="$(dig "$NC_DOMAIN" A +short | grep -P "^[0-9.]+$" | head -n1)"
+fi
+if [ -n "$(dig "$NC_DOMAIN" AAAA +short | grep -P "^[0-9a-fA-F:]+$" | head -n1)" ]; then
+    IPv6_ADDRESS_NC="$(dig "$NC_DOMAIN" AAAA +short | grep -P "^[0-9a-fA-F:]+$" | head -n1)"
+fi
```

Comment:
why do we need to whitelist the NC_DOMAIN ip-address now? That was formerly not necessary, right?

Reply:
If it wasn't necessary before, it shouldn't be now. Blacklist: "recommended" is the same list of blocked CIDRs as you had in your coturn configuration before.

Reply:
All right, thanks!

Reply:
it is needed for the relay option

Reply:
But why do we do this now when it worked without before?

Reply:
the relay option seems to require the public ips, coturn didn't required it

Reply:
I see, you might be right: https://github.com/coturn/coturn/blob/8b802f9a5e308e3dd3434ed403bb79c4fbcdbc10/README.turnserver#L373-L398

But cannot we fall-back to this?
![image](https://github.com/nextcloud/all-in-one/assets/42591237/2c4699b8-93cc-4400-bd83-0f44de93edb1)

Reply:
but we need the ip the users connect to, not the ip the server uses to connect to the internet?

Reply:
I thought it needs the public ip-address  of the server as relady-address which should be discovered automatically via this feature IIRC?

Reply:
Maybe I'm wrong, but I think we need the IP over which the server is reachable (A/AAAA records) and the inbuilt check gives us the IP the servers uses to connect to the internet

Reply:
> Maybe I'm wrong, but I think we need the IP over which the server is reachable (A/AAAA records) and the inbuilt check gives us the IP the servers uses to connect to the internet

But isnt that usually the same ip-address?

Reply:
but not in all cases

Reply:
> but not in all cases

In which for example? Can you make an example?

Reply:
yes:
My reverse proxy is a server (for example hosted at netcup).
AIO runs on a server at my home.
there is a vpn between both, which proxies 192.168.168.0/24.
homeserver: 192.168.168.2
netcup server: 192.168.168.1
the reverse proxie connects to 192.168.168.2:11000
I use iptables port forwarding to port forward the tcp/udp ports of talk
Now when I will open my cloud i need the ip of the netcup server and if I ping 1.1.1.1 from my talk container, I ping it with my home ip

Reply:
Ah I see. But is this a setup we actually want to support? I had thr feeling that many people that run sich a setup dont even know how to do that:

> I use iptables port forwarding to port forward the tcp/udp ports of talk

Reply:
Yes that was maybe a case that is very unrealistic, but iptables port forwarding normally works
another case would be, if the server has multiple ips and the server connects to the internet over another ip then clients connect to the cloud

Reply:
> Yes that was maybe a case that is very unrealistic

Ttue

> another case would be, if the server has multiple ips and the server connects to the internet over another ip then clients connect to the cloud

This is acutally a good example. 

However I am concerned about setups with a dynamic public ip-address. Only a container restart will then add this new ip-address. In contrast the automatic logic might automatically use the latest public ip-address directly?

Reply:
I don't know if eturnal fetches the ip at starup or in a intervall, lets wait what sando says

Reply:
> I don't know if eturnal fetches the ip at starup or in a intervall, lets wait what sando says

If it fetches at startup then it makes no difference. If it fetches regularly then it does

Reply:
> another case would be, if the server has multiple ips and the server connects to the internet over another ip then clients connect to the cloud

yes, that is a good example. This is a case where the advertised [relay_ipv4_addr](https://eturnal.net/documentation/#relay_ipv4_addr) can hardly be auto-detected. So, I would go with the custom mounted configuration file in that case as I assume that your "80%" users do not have that use case.

Reply:
> However I am concerned about setups with a dynamic public ip-address. Only a container restart will then add this new ip-address.

eturnal does not need a restart, but it must be made aware of that. Covering (dynamic) IP addresses changes requires the [relay_ipv4_addr](https://eturnal.net/documentation/#relay_ipv4_addr) to be set in a configuration file **if** you want to avoid a container restart. 

Basically, a script checks if the IP address has changed, changes the respective address with e.g. `sed` in the configuration file and then triggers an `eturnalctl reload`. This will not lead to any interruption of the TURN service's active sessions/calls, and it does not do anything if no changes in configuration exists. We have an example script in the [repository](https://github.com/processone/eturnal/blob/master/scripts/examples/check-relay-ip). This may be integrated with supervisor. 

>  In contrast the automatic logic might automatically use the latest public ip-address directly without a restart needed?

The "automatic logic" detects the address at startup and continues to use it until configured differently or restarted.

Reply:
also for me again: needs the relay ip to be the ip users connect to or the ip the server connects to the internet?

Reply:
It is the IP address users connect to. And in the case that the TURN server resides on a different host then than the main nextcloud, it is the IP address of the host, where the TURN server is located - unless there is some routing activities e.g. with a reverse proxy..

Reply:
https://github.com/nextcloud/all-in-one/pull/2855#discussion_r1241233958